### PR TITLE
Misc cleanup in extensions

### PR DESCRIPTION
**Pull request by rsvoboda**
Misc cleanup in extensions

* 14e889dbb7 - Avoid deprecated clazz.newInstance() (10 minutes ago) <Rostislav Svoboda>
* 2ada914443 - Use Objects.requireNonNullElseGet to simplify null checks (26 minutes ago) <Rostislav Svoboda>
* c5ff22f89f - Use Map.forEach or Map.values (45 minutes ago) <Rostislav Svoboda>
* b08e451c9b - Use directly .sort instead of Collections.sort wrapper (61 minutes ago) <Rostislav Svoboda>
* 17c61153f1 - Use enhanced for loop (83 minutes ago) <Rostislav Svoboda>
* f6d1a0df80 - Simplified if-clauses (86 minutes ago) <Rostislav Svoboda>
* f0ded802a8 - Unused imports (89 minutes ago) <Rostislav Svoboda>

### CODE REVIEW

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/MethodValidatedAnnotationsTransformer.java [hunk 1]
```diff
@@ -101,9 +101,7 @@ private boolean isJaxrsMethod(MethodInfo method) {
         // check superclass, but only the direct one since we do not (yet) have the entire ClassInfo hierarchy here
         DotName superClass = clazz.superName();
         if (!superClass.equals(IndexingUtil.OBJECT)) {
-            if (isJaxrsMethod(signatureKey, superClass)) {
-                return true;
-            }
+            return isJaxrsMethod(signatureKey, superClass);
```
Comment (gsmet) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/MethodValidatedAnnotationsTransformer.java:104]: Please keep it the way it was. I agree it leads to the same result but the logic is slightly different and I prefer we keep it that way.

File: extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java [hunk 2]
```diff
@@ -68,13 +69,8 @@ public DataSourceBuildTimeConfig getDataSourceRuntimeConfig(String dataSourceNam
...
             return defaultDataSource;
         }
-
         DataSourceBuildTimeConfig dataSourceBuildTimeConfig = namedDataSources.get(dataSourceName);
-        if (dataSourceBuildTimeConfig == null) {
-            return new DataSourceBuildTimeConfig();
-        }
-
-        return dataSourceBuildTimeConfig;
+        return Objects.requireNonNullElseGet(dataSourceBuildTimeConfig, DataSourceBuildTimeConfig::new);
```
Comment (gsmet) [inline extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java:73]: Can't say I find this more readable but I suppose it's a matter of taste.

**Review by gsmet — CHANGES_REQUESTED** [review]
I had a look, looks good overall except for one change I would prefer we keep it the way it is (and it's in my code so I really would prefer we stick to the current form).

**Comment by gsmet** [comment]
> fyi - quarkus-bot notified just one person while it added several component labels. IMHO it should notify more people ;).

Nope. Notifications in PR is a different feature that is enabled for very few components. I let this for the component owners to decide if they want it or not.

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/MethodValidatedAnnotationsTransformer.java (hunk 1 above)
Comment (rsvoboda) [reply extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/MethodValidatedAnnotationsTransformer.java]: change reverted

File: extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java (hunk 2 above)
Comment (rsvoboda) [reply extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java:73]: I like the compact form, agree that the naming is not ideal though

**Comment by rsvoboda** [comment]
PR is updated

**Comment by rsvoboda** [comment]
I can't reproduce  smallrye-reactive-messaging-kafka failure on my local env

Running `mvn clean install -pl extensions/smallrye-reactive-messaging-kafka/deployment -Dtest-containers -Dstart-containers -am -Dtest=KafkaDevServicesDevModeTestCase -DfailIfNoTests=false`

**Comment by rsvoboda** [comment]
Rebased to pick the latest changes, Clement mentioned he did some changes yesterday

**Review by gsmet — APPROVED** [review]

**Pull request merged**